packages/cli/src/util/build/write-build-result.ts

```diff
@@ -156,11 +156,18 @@ async function writeBuildResultV2(
         const ext = getFileExtension(fallback);
         const fallbackName = `${normalizedPath}.prerender-fallback${ext}`;
         const fallbackPath = join(outputDir, 'functions', fallbackName);
-        const stream = fallback.toStream();
-        await pipe(
-          stream,
-          fs.createWriteStream(fallbackPath, { mode: fallback.mode })
-        );
+
+        // if we're in the build container and fallback is already
+        // on the disk just move it to avoid duplicating output
+        if (process.env.VERCEL && 'fsPath' in fallback) {
```

Comment:
Are you sure `process.env.VERCEL` will work as you expected.

Seems like `vc pull && vc build` will assign that env var locally when the user has System Env Vars checkbox checked.

Reply:
Is there a better env variable you are aware of we could rely on instead?

Reply:
Checking for `VERCEL` is indeed not going to be reliable. It's set when running `vc build` locally as well. IMO we should introduce an env var in build-container. We could also check the execution path, i.e. build-container has a root-level `/vercel` directory.

Reply:
In general though, I've tried to avoid logic like this in `vc build`. Instead, we should add a CLI flag (or env var) which alters the behavior of `vc build`. For example, build-container sets the `--output` flag to specify where the BOA files should go, rather then having that logic baked directly in to `vc build`.